packages/wallets/particlenetwork/src/adapter.ts

```diff
@@ -0,0 +1,178 @@
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+
+    constructor(config: ParticleNetworkAdapterConfig) {
+        super();
+        this._connecting = false;
+        this._publicKey = null;
+        this._wallet = null;
+        this._config = config;
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._wallet?.auth.isLogin();
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            let wallet: ParticleNetwork;
+            try {
+                wallet = window.particle || new ParticleNetwork(this._config);
+            } catch (error: any) {
+                throw new WalletConfigError(error?.message, error);
+            }
+
+            let userInfo;
+            try {
+                userInfo = await wallet.getSolanaProvider().connect();
+            } catch (error: any) {
+                throw new WalletConnectionError(error?.message, error);
+            }
+
+            if (!userInfo.publicKey) {
+                throw new WalletAccountError();
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(userInfo.publicKey);
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                if (wallet.auth.isLogin()) await wallet.getSolanaProvider().disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async signTransaction(transaction: Transaction): Promise<Transaction> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            try {
+                const tx = await wallet.getSolanaProvider().signTransaction(bs58.encode(transaction.serialize()));
+                return Transaction.from(Buffer.from(tx, 'base64'));
```

Comment:
This looks pretty strange -- you're encoding the serialized tx as base58, but deserializing it from base64. Using base64 everywhere (except for string signature values) will be faster. Also, please don't use `Buffer`, which is a Node.js class. Instead, refactor to use `Uint8Array`.